### DISCUSSION TITLE
What does "start_by_longest" exactly do ?

### DISCUSSION THREAD

**Discussion by Ca-ressemble-a-du-fake** [coqui-ai/TTS#2016] [General Q&A] [answered]
Hi,

For the same dataset I was sometimes having OOM right in the beginning but not always. I could train it up to 10 k steps until it crashed for a different (not understood) reason, and then I upgraded everything from drivers (515.65) to Ubuntu (22.04) to Cuda (11.7), and it crashed after ~ 50 steps. Before those crashes `nvidia-smi` showed a varying memory usage.

Now that I run the training with `start_by_longest` I see that the memory usage (and the power consumption) stays longer at about ~ 23 / 24 GB (see RTX 3090) but sometimes goes very near to the limit. 

The comment in code states 
> start_by_longest (bool): Start by longest sequence. It is especially useful to check OOM. Defaults to False.

I understand from that piece of information that if the first batch runs fine then the following iterations will run also fine. But `nvidia-smi` can show growing values, so I am not sure it won't hit an OOM further in the training. Is it actually OOM proof (if it does not fail from start) ?

Consequently does `start_by_longest` uses the longest samples at the beginning of the training or only at the beginning of each batch round (round meaning step) ?

_I don't know if it makes a difference but I am always running the training with a custom config.json (not a recipe) because it is easier to store the config file alongside the dataset._

Thank you in advance for explaining this

**Comment by erogol** [comment, marked as answer]
Starts from the longest sample in the dataset so it allocates the most VRAM used in the training.
